### PR TITLE
Optimization in ObserveCompositeRelationFilter

### PR DESCRIPTION
**Pull request by Michal-Wadowski**
Reduce decoding overhead with passing path list via user context in ObserveCompositeRelationFilter

### CODE REVIEW

File: leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ObserveUtil.java [hunk 1]
```diff
@@ -133,17 +147,27 @@ public ObserveCommon(Request request) {
...
+        return context;
+    }
+
+    /**
+     * Update user context with encoded list of {@link LwM2mPath}.
+     * 
+     * @param context user context
+     * @param paths the list of {@link LwM2mPath}
+     */
+    public static void putPathsIntoContext(Map<String, String> context, List<LwM2mPath> paths) {
```
Comment (sbernard31) [inline leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ObserveUtil.java:164]: Maybe `addPathstoContext()` ? (if we keep `putPathsIntoContext` it's OK too)

File: leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ObserveUtil.java [hunk 2]
```diff
@@ -133,17 +147,27 @@ public ObserveCommon(Request request) {
         context.put(CTX_ENDPOINT, endpoint);
         context.put(CTX_REGID, registrationId);
 
+        List<LwM2mPath> paths = request.getPaths();
+
+        putPathsIntoContext(context, paths);
```
Comment (sbernard31) [inline leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ObserveUtil.java:152]: WDYT about ? 

```suggestion
        putPathsIntoContext(context, request.getPaths());
```

Comment (Michal-Wadowski) [reply leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/ObserveUtil.java:152]: No problem :smile:

**Comment by sbernard31** [comment]
Thx @Michal-Wadowski  :pray:

**Pull request merged**